src/main/kotlin/com/pubnub/api/subscribe/eventengine/effect/effectprovider/HandshakeProviderImpl.kt

```diff
@@ -1,19 +1,22 @@
 package com.pubnub.api.subscribe.eventengine.effect.effectprovider
 
+import com.pubnub.api.PubNub
 import com.pubnub.api.endpoints.pubsub.Subscribe
 import com.pubnub.api.endpoints.remoteaction.RemoteAction
+import com.pubnub.api.endpoints.remoteaction.map
 import com.pubnub.api.subscribe.eventengine.event.SubscriptionCursor
 
-internal class HandshakeProviderImpl(val subscribe: Subscribe) : HandshakeProvider {
+internal class HandshakeProviderImpl(val pubNub: PubNub) : HandshakeProvider {
 
     override fun getHandshakeRemoteAction(
         channels: List<String>,
         channelGroups: List<String>
     ): RemoteAction<SubscriptionCursor> {
+        val subscribe = Subscribe(pubNub)
         subscribe.channels = channels
         subscribe.channelGroups = channelGroups
         subscribe.timetoken = 0
         subscribe.region = null
-        return RemoteActionForHandshake(subscribe)
+        return subscribe.map { SubscriptionCursor(it.metadata.timetoken, it.metadata.region) }
```

Comment:
Is there anything that could make it more readable?

Reply:
For me those separate classes like RemoteActionForReceiveMessagesResult.kt clearly stated what code is doing but that maybe is related to my lack of experience in such abstractions.

What might improve code in the current shape is renam of `map` class to `mapTo..... `This is because map might be confuse with standard kotlin `map` function

Reply:
e.g. mapToExtendedRemoteAction.

Reply:
Will `mapResult` be fine with you?

Reply:
Choosing between `map` and `mapResult` I would go with map.

Reply:
`map` that is